src/components/select/select.tsx

```diff
@@ -255,11 +255,28 @@ export class Select {
     ) {
         event.stopPropagation();
 
+        const selector = `#${this.portalId} limel-menu-surface limel-list`;
+        const list: HTMLElement = document.querySelector(selector);
+
+        if (!list) {
+            return;
+        }
+
+        const scrollPosition = list.scrollTop;
+        const delay = 8;
+
+        list.scrollTop += 10; // Increase the scroll position by 10
+
         if (isMultiple(event.detail)) {
             const listItems: ListItem[] = event.detail;
             const options: Option[] = listItems.map((item) => item.value);
             this.change.emit(options);
 
+            // Use setTimeout to restore scroll position after a slight delay
+            setTimeout(() => {
+                list.scrollTop = scrollPosition;
+            }, delay);
```

Comment:
Using any delay other than 0 introduces a race condition. Either this works with a delay of 0, or we can't rely on this solution as a fix, because it depends on a race condition. (Although, in this case, if it usually works, and only sometimes doesn't, that's a better behaviour for the user than the current behaviour.)

I assume you've already tried this with a delay of 0, but I'll try it myself just in case, and if it doesn't work, I might play around and see if I can get it to work in some other way.

Reply:
> (Although, in this case, if it usually works, and only sometimes doesn't, that's a better behaviour for the user than the current behaviour.)

I agree!

> I assume you've already tried this with a delay of 0, but I'll try it myself just in case, and if it doesn't work, I might play around and see if I can get it to work in some other way.

Thank you, Adrian! 🙏

Reply:
⚡

Reply:
Please help test my fixup.

It's a pity that John isn't here to help test it, since he's the one who was having problems with the previous implementation… 😕

Reply:
I am actually reading the thread because I'm curious to see if your fixup will solve it. When I still had the issues, it was when I had selected multiple items, but it wasn't consistent. I tried the requestAnimationFrame but didn't try the setTimeout within the rAF as you did. :D 


I'll test it now. It's too hot to be outside so I'm in the living room with a fan a full blast and reading through issues in the cooldown board 😆